feature-libs/product-configurator/rulebased/components/add-to-cart-button/configurator-add-to-cart-button.component.html

```diff
@@ -4,6 +4,24 @@
       <button
         class="cx-btn btn btn-block btn-primary cx-add-to-cart-btn"
         (click)="onAddToCart(container.configuration, container.routerData)"
+        [attr.aria-label]="
+          (getButtonResourceKey(container.routerData, container.configuration)
+            | cxTranslate) +
+          ' ' +
+          ('configurator.a11y.addToCartPrices'
+            | cxTranslate
+              : {
+                  basePrice:
+                    container.configuration?.priceSummary?.basePrice
+                      ?.formattedValue,
+                  selectedOptions:
+                    container.configuration?.priceSummary?.selectedOptions
+                      ?.formattedValue,
+                  totalPrice:
+                    container.configuration?.priceSummary?.currentTotal
+                      ?.formattedValue
+                })
```

Comment:
this is a quite complex inline expression. 
maybe one can ease it extracting line 13 to 23 into an controller method. Then you can also test more easily all the different possible paths. 
```
  extractConfigPrices(configuration: Configurator.Configuration) {
    let priceSummary = configuration?.priceSummary;
    return {
      basePrice: priceSummary?.basePrice?.formattedValue,
      selectedOptions: priceSummary?.selectedOptions?.formattedValue,
      totalPrice: priceSummary?.currentTotal?.formattedValue,
    };
```
@ChristophHi @Schuppell any thoughts on this

Reply:
Yes, I agree -> probably better to have the accesses for getting the formatted values in the controller

Reply:
done, please check.